src/Adapters/PageIdsRepo.php

```diff
@@ -30,4 +30,15 @@ public function getPageIdsOfPagesWithoutPersistentIds( int $limit ): array {
 		);
 	}
 
+	public function getPageIdFromPersistentId( string $persistentId ): ?int {
+		$row = $this->database->newSelectQueryBuilder()
+			->select( 'p.page_id' )
+			->from( 'page', 'p' )
+			->join( 'persistent_page_ids', 'ppi', 'p.page_id = ppi.page_id' )
+			->where( [ 'ppi.persistent_id' => $persistentId ] )
```

Comment:
Why are we doing a join with the page table?

Isn't this enough?

```sql
SELECT page_id FROM persistent_page_ids WHERE persistent_id = foo
```

Same for the pre-existing `getPageIdsOfPagesWithoutPersistentIds`

Reply:
@malberts

Reply:
Btw, the new method is better placed in `PersistentPageIdentifiersRepo` and its implementations.

`PageIdsRepo` is a special lookup service that should not be named the way it is, since it does not follow the Repository pattern.

Reply:
`getPageIdsOfPagesWithoutPersistentIds` needs it. We're looking for pages without persistent IDs. Those page IDs do not exist in the `persistent_page_ids` table, so we need to check MW's `page` table.

Reply:
But you're right. `getPageIdFromPersistentId` just needs to query our table. If a page has a persistent ID, then it will be in the table.

Reply:
I am unsure now if the complicated query started out because we needed more information from the page table at some point. Or if it had something to do with some condition where a page might no longer exist in the page table, and since we don't delete rows from our table, the queries avoid returning data for missing pages. I will need to look at this again.

Reply:
If it is deliberate, then that is the kind of thing that warrants a code comment.

Reply:
Regarding the existing join: https://github.com/ProfessionalWiki/PersistentPageIdentifiers/pull/69

In the case of this PR the join is indeed not necessary. Basically the new query without a join will get the page ID for a deleted page, but the redirect is blocked later when it checks page existence.